plugins/camera/camera.h

```diff
@@ -0,0 +1,381 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+#include <map>
+#include "plugin_base.h"
+
+namespace dronecore {
+
+class CameraImpl;
+class System;
+
+/**
+ * @brief The Camera class can be used to manage cameras that implement the
+ * MAVLink Camera Protocol: https://mavlink.io/en/protocol/camera.html.
+ *
+ * The plugin needs to be instantiated multiple times for multiple cameras.
+ * However, this is not supported yet.
+ *
+ * Synchronous and asynchronous variants of the camera methods are supplied.
+ */
+class Camera : public PluginBase
+{
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto camera = std::make_shared<Camera>(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit Camera(System &system);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~Camera();
+
+    /**
+     * @brief Possible results returned for camera commands.
+     */
+    enum class Result {
+        SUCCESS = 0,
+        IN_PROGRESS,
+        BUSY,
+        DENIED,
+        ERROR,
+        TIMEOUT,
+        WRONG_ARGUMENT,
+        UNKNOWN
+    };
+
+    /**
+     * @brief Returns a human-readable English string for Camera::Result.
+     *
+     * @param result The enum value for which a human readable string is required.
+     * @return Human readable string for the Camera::Result.
+     */
+    static const char *result_str(Result result);
+
+    /**
+     * @brief Callback type for asynchronous Camera calls.
+     */
+    typedef std::function<void(Result)> result_callback_t;
+
+    /**
+     * @brief Take photo (synchronous).
+     *
+     * This takes one photo immediately.
+     *
+     * @return Result of request.
+     */
+    Result take_photo();
+
+    /**
+     * @brief Start photo interval (synchronous).
+     *
+     * Starts a photo timelapse given an interval.
+     *
```

Comment:
Maybe add see also to the corresponding stop method (and visa versa).

```
* @sa stop_photo_interval
```

Reply:
It can however be confusing because it would mean that the `async` versions go together as well as the sync versions but that's not the case and they can be mixed and matched.